src/modules/pivotTable/applyLegendSet.js

```diff
@@ -0,0 +1,68 @@
+import { isColorBright } from './isColorBright'
+
+const LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM = 'BY_DATA_ITEM'
+const LEGEND_DISPLAY_STRATEGY_FIXED = 'FIXED'
+
+const LEGEND_DISPLAY_STYLE_FILL = 'FILL'
+const LEGEND_DISPLAY_STYLE_TEXT = 'TEXT'
+
+const valueFitsLegend = (value, legend) =>
+    legend.startValue <= value && legend.endValue > value // TODO: Confirm inclusive/exclusive bounds
+
+const getLegendSet = engine => {
+    let legendSetId
+    switch (engine.visualization.legendDisplayStrategy) {
+        case LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM:
+            legendSetId = undefined // TODO: ByDXID LegendSet
+            //engine.rawData.metaData.items[dxId].legendSet
```

Comment:
Can we have multiple different legendSets at the same time?
If we put the fetch of the legendSet by dx id in the wrapper plugin, here we would just need to use whatever legend set is passed in.

Reply:
Yeah there can be multiple different legendSets - there might be multiple DX items which each has its own legendSet assigned, so we need to look up the relevant legendSet based on the dx of whichever cell we're in.